### ISSUE TITLE
Incompatible with matplotlib 3

### ISSUE THREAD

**Issue by djhoese** [SciTools/cartopy#1126]
### Description
It seems some small changes in matplotlib 3.x are causing exceptions in cartopy. The commit in matplotlib that broke things in cartopy is:

https://github.com/matplotlib/matplotlib/commit/86fe16bfe7ed27985457637356f42b0ff98bdaca#diff-43183989add1c60a82728118bd9efe98R2714

The issue is that the `img` data in `imshow` is no longer passed as a positional argument but rather as the `X` keyword argument.

#### Code to reproduce

```
In [1]: from cartopy import crs

In [2]: import matplotlib.pyplot as plt

In [3]: import numpy as np

In [4]: ax = plt.axes(projection=crs.PlateCarree())

In [5]: plt.imshow(np.zeros((100, 100)))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-b314fa645876> in <module>()
----> 1 plt.imshow(np.zeros((100, 100)))

~/anaconda/envs/polar2grid_py36/lib/python3.6/site-packages/matplotlib/pyplot.py in imshow(X, cmap, norm, aspect, interpolation, alpha, vmin, vmax, origin, extent, shape, filternorm, filterrad, imlim, resample, url, data, **kwargs)
   2651         vmax=vmax, origin=origin, extent=extent, shape=shape,
   2652         filternorm=filternorm, filterrad=filterrad, imlim=imlim,
-> 2653         resample=resample, url=url, data=data, **kwargs)
   2654     sci(__ret)
   2655     return __ret

TypeError: imshow() missing 1 required positional argument: 'img'
```

#### Traceback 

```
  File "/Users/davidh/anaconda/envs/polar2grid_py36/lib/python3.6/site-packages/matplotlib/pyplot.py", line 2653, in imshow
    resample=resample, url=url, data=data, **kwargs)
TypeError: imshow() missing 1 required positional argument: 'img'
```

<details>
 <summary>Full environment definition</summary>

### conda list

```
```

### pip list

```
```

</details>

**Comment by djhoese** [comment]
NOTE: This is only a problem if you use the `pyplot` interface.

**Comment by djhoese** [comment]
CC'ing @anntzer and @jklymak as the committer and merger of the matplotlib change that affected this. To the two of you: is there a way to not pass `X` by keyword but only by position to the Axes object's `imshow` method?

**Comment by jklymak** [comment]
I don't think I had anything to do w/ that PR :wink:

**Comment by djhoese** [comment]
Sorry @jklymak. You're right. I looked at the new PRs that were merged after the associated commit in the commit history. I can't find a PR for this commit. Maybe it was done with a direct push?

**Comment by djhoese** [comment]
Found it: https://github.com/matplotlib/matplotlib/pull/10918
Looks like @anntzer merged it themselves.

**Comment by dopplershift** [comment]
I think this is fixed by matplotlib 3.0.1.
